test_util/integration_test.py

```diff
@@ -1290,75 +1292,105 @@ def wait_for_endpoint():
 
     r = requests.get(test_cache_url)
 
+    # Handy for diagnosing strange signal service test behavior, not sure if we should
+    # leave these in for master branch, or remove for the final PR.
+    print('TESTING SIGNAL RETURN:\n{}'.format(r.text))
+    print('CACHE SERVER STATUS:\n{}'.format(r.status_code))
+
     r_data = json.loads(r.json())
 
     cluster.destroy_marathon_app(signal_app_definition['id'])
     cluster.destroy_marathon_app(test_server_app_definition['id'])
 
     exp_data = {
-            'Event': 'health',
-            'UserId': '',
-            'ClusterId': '',
-            'Properties': {
-                'provider': 'onprem',
-                'source': 'cluster',
-                'clusterId': '',
-                'customerKey': '',
-                'environmentVersion': '',
-                'variant': 'open'}
-            }
+        'diagnostics': {
+            'event': 'health',
+            'anonymousId': 'open /var/lib/dcos/cluster-id: no such file or directory',
```

Comment:
We purposefully had the error propagate into the signal service like that so we can see what errored out, since there are a few type of errors we can have loading the cluster-id file, null could indicate any one of those errors and we want to surface the actual problem in segment. 

There isn't an option to change the location of the cluster-id currently, since this is a well established file, whose location does not change. However, if we did have such an option would you like another marathon app to make a fake cluster-id on the agent? What would the test look like in that case? And would that test actually test the signal service in the way we expect? 

My opinion is that the signal service should be tested on a master, and that it should return a not empty string for cluster-id as we expect, but we simply can't do that right now.